src/main/java/com/dev/MyThread.java

```diff
@@ -0,0 +1,20 @@
+package com.dev;
+
+import org.apache.log4j.Logger;
+
+public class MyThread extends Thread {
+    private static final Logger logger = Logger.getLogger(MyThread.class);
+    private Counter counter;
+
+    public MyThread(Counter counter) {
+        this.counter = counter;
+    }
+
+    @Override
+    public void run() {
+        for (int i = 0; i < 20; i++) {
```

Comment:
```suggestion
       while (counter.getCounter() < 20) {
```
you shouldn't do it 20 times, but until the counter will reach a certain value